src/lib/operations.ts

```diff
@@ -122,6 +126,120 @@ const prepareToEditExtrude: PrepareToEditCallback =
     }
   }
 
+/**
+ * Gather up the argument values for the Chamfer or Fillet command
+ * to be used in the command bar edit flow.
+ */
+const prepareToEditEdgeTreatment: PrepareToEditCallback = async ({
+  operation,
+  artifact,
+}) => {
+  const isChamfer =
+    artifact?.type === 'edgeCut' && artifact.subType === 'chamfer'
+  const isFillet = artifact?.type === 'edgeCut' && artifact.subType === 'fillet'
+  const baseCommand = {
+    name: isChamfer ? 'Chamfer' : 'Fillet',
+    groupId: 'modeling',
+  }
+  if (
+    operation.type !== 'StdLibCall' ||
+    !operation.labeledArgs ||
+    (!isChamfer && !isFillet)
+  ) {
+    return { reason: 'Wrong operation type or artifact' }
+  }
+
+  // Recreate the selection argument (artiface and codeRef) from what we have
+  const edgeArtifact = getArtifactOfTypes(
+    {
+      key: artifact.consumedEdgeId,
+      types: EDGE_TREATMENT_TYPES,
+    },
+    engineCommandManager.artifactGraph
+  )
+  if (err(edgeArtifact)) {
+    return { reason: "Couldn't find edge artifact" }
+  }
+
+  let edgeCodeRef = getEdgeCutConsumedCodeRef(
+    artifact,
+    engineCommandManager.artifactGraph
+  )
+  if (err(edgeCodeRef)) {
+    return { reason: "Couldn't find edge coderef" }
+  }
+  const selection = {
+    graphSelections: [
+      {
+        artifact: edgeArtifact,
+        codeRef: edgeCodeRef,
+      },
+    ],
+    otherSelections: [],
+  }
+
+  // Assemble the default argument values for the Fillet command,
+  // with `nodeToEdit` set, which will let the Fillet actor know
+  // to edit the node that corresponds to the StdLibCall.
+  const nodeToEdit = getNodePathFromSourceRange(
+    kclManager.ast,
+    sourceRangeFromRust(operation.sourceRange)
+  )
+  const isPipeExpression = nodeToEdit.some(
+    ([_, type]) => type === 'PipeExpression'
+  )
+  if (!isPipeExpression) {
+    return {
+      reason: 'Only chamfer in pipe expressions are supported for edits',
+    }
+  }
```

Comment:
This is our new guard for now @Irev-Dev. I'm honestly thinking we should merge as is, and try to expand later. Got too many branches at once for these flows and I'm afraid I'll get lost haha.